packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -261,19 +261,26 @@ class CliServiceProvider extends ServiceProviderCore implements ServiceProvider
     const topology = this.getTopology() as Topology;
     const { version } = require('../package.json');
     let cmdLineOpts = null;
-    try {
-      cmdLineOpts = await this.runCommandWithCheck('admin', {
-        getCmdLineOpts: 1
-      }, this.baseCmdOptions);
-      // eslint-disable-next-line no-empty
-    } catch (e) {
-    }
+    let atlasVersion = null;
+    await Promise.all([
+      (async() => {
+        cmdLineOpts = await this.runCommandWithCheck('admin', {
+          getCmdLineOpts: 1
+        }, this.baseCmdOptions).catch(e => e);
+      })(),
+      (async() => {
+        atlasVersion = await this.runCommandWithCheck('admin', {
+          atlasVersion: 1
+        }, this.baseCmdOptions).catch(e => e);
+      })()
+    ]);
```

Comment:
Not sure how much this changes the behavior of the code that relies on those variables, but before `cmdLineOpts` would not be re-assigned if any error happened, but now by returning error from `catch` it can be an error, right? Should we noop those `catch` calls instead, what do you think?

Also maybe we can do this part using destructuring so there is no need for IIFE (feels a bit more readable that way), how does it look to you?

```suggestion
    const [cmdOptions = null, atlasVersion = null] = await Promise.all([
      this.runCommandWithCheck('admin', { getCmdLineOpts: 1 }, this.baseCmdOptions).catch(() => {}),
      this.runCommandWithCheck('admin', { atlasVersion: 1 }, this.baseCmdOptions).catch(() => {})
    ])
```

Reply:
Yeah, I like your suggestion a lot :+1: